sky/skylet/providers/lambda_labs/node_provider.py

```diff
@@ -0,0 +1,195 @@
+        providers are allowed to cache the result of a call to
+        non_terminated_nodes() to serve single-node queries
+        (e.g. is_running(node_id)). This means that non_terminated_nodes() must
+        be called again to refresh results.
+
+        Examples:
+            >>> provider.non_terminated_nodes({TAG_RAY_NODE_KIND: "worker"})
+            ["node-1", "node-2"]
+        """
+        nodes = self._get_filtered_nodes(tag_filters=tag_filters)
+        return [k for k, _ in nodes.items()]
+
+    def is_running(self, node_id):
+        """Return whether the specified node is running."""
+        return self._get_node(node_id=node_id) is not None
+
+    def is_terminated(self, node_id):
+        """Return whether the specified node is terminated."""
+        return self._get_node(node_id=node_id) is None
+
+    def node_tags(self, node_id):
+        """Returns the tags of the given node (string dict)."""
+        return self._get_cached_node(node_id=node_id)['tags']
+
+    def external_ip(self, node_id):
+        """Returns the external ip of the given node."""
+        return self._get_cached_node(node_id=node_id)['external_ip']
+
+    def internal_ip(self, node_id):
+        """Returns the internal ip (Ray ip) of the given node."""
+        return self._get_cached_node(node_id=node_id)['internal_ip']
+
+    def create_node(self, node_config, tags, count):
+        assert count == 1, count   # Only support 1-node clusters for now
+        self._create_node(node_config, tags, count)
+
+    def _create_node(self, node_config, tags, count):
+        """Creates a number of nodes within the namespace."""
+        del count  # unused
+
+        # get the tags
+        config_tags = node_config.get('tags', {}).copy()
+        config_tags.update(tags)
+        config_tags[TAG_RAY_CLUSTER_NAME] = self.cluster_name
+
+        # create the node
+        ttype = node_config['InstanceType']
+        region = self.provider_config['region']
+        vm_resp = self.lambda_client.up(instance_type=ttype,
+                                        region=region,
+                                        quantity=1,
+                                        name=self.cluster_name)
+        vm_list = vm_resp.get('data', []).get('instance_ids', [])
+        assert len(vm_list) == 1, len(vm_list)
+        vm_id = vm_list[0]
+        self.metadata[vm_id] = {'tags': config_tags}
+
+        # Wait for booting to finish
+        # TODO(ewzeng): For multi-node, launch all vms first and then wait.
+        while True:
+            vms = self.lambda_client.ls().get('data', [])
+            for vm in vms:
+                if vm['id'] == vm_id and vm['status'] == 'active':
+                    # Hack: send a copy of metadata file to the head node.
+                    # This is because autodown runs `ray up` and `ray down`
+                    # there.
+                    # TODO(ewzeng): remove when Lambda Labs adds tag support.
```

Comment:
Might be worth letting the Lambda folks know that this would be useful for SkyPilot (and probably in general) if you haven't already.

Reply:
I did! But maybe I should bug them more about it :)